bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/AbstractNetatmoThingHandler.java

```diff
@@ -233,21 +237,30 @@ protected void updateProperties(Integer firmware, String modelId) {
         updateProperties(properties);
     }
 
+    protected Optional<RadioHelper> getRadioHelper() {
+        RadioHelper helper = radioHelper;
+        return helper != null ? Optional.of(helper) : Optional.empty();
+    }
+
+    protected Optional<BatteryHelper> getBatteryHelper() {
+        BatteryHelper helper = batteryHelper;
+        return helper != null ? Optional.of(helper) : Optional.empty();
+    }
+
     public void updateMeasurements() {
     }
 
-    public void getMeasurements(String device, @Nullable String module, String scale, List<String> types,
+    public void getMeasurements(@Nullable String device, @Nullable String module, String scale, List<String> types,
             List<String> channels, Map<String, Float> channelMeasurements) {
-        NetatmoBridgeHandler handler = getBridgeHandler();
-        if (handler == null) {
+        if (!getBridgeHandler().isPresent() || device == null) {
```

Comment:
Caching the bridge handler result in a local variable is much safer from a concurrency standpoint.